packages/protocol-kit/src/contracts/BaseContract.ts

```diff
@@ -80,24 +98,68 @@ class BaseContract<ContractAbiType extends InterfaceAbi & Abi> {
   }
 
   async init() {
-    this.contract = new Contract(
-      this.contractAddress,
-      this.contractAbi,
-      (await this.safeProvider.getExternalSigner()) || this.runner
-    )
+    this.wallet = await this.safeProvider.getExternalSigner()
+    this.contract = getContract({
+      address: asAddress(this.contractAddress),
+      abi: this.contractAbi,
+      client: this.wallet || this.runner
+    })
+  }
+
+  async getTransactionReceipt(hash: Hash) {
+    const client = this.runner
+    return client?.getTransactionReceipt({ hash })
+  }
+
+  async convertOptions(
```

Comment:
Why is this conversion necessary. Would be helpful to have some comments / jsdoc here or in the implementations inside the protocol-kit utils

Reply:
The conversion is necessary because the `writeContract` parameters takes a `TransactionRequest` type (with some omitted properties that can be inferred from the `Contract` instance. Viem does discriminate between transaction types. There is a specific type introduced post-london hardfork that changed some of the fee fields. What they do is that they create a union between a `BaseTransactionRequest` and the different fee properties as defined here: `https://github.com/wevm/viem/blob/viem%402.18.0/src/types/fee.ts`. Our `TransactionOptions` is somehow a more permissable one while their types are mutually exclusive. Our `TransactionOptions` are used in multiple places so I didnt want to break the consistency between the interface (and that its even destructured into the `TransactionResult` which is also all over the place. I added a comment to address that on the function as suggested.